src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/ArmorTask.java

```diff
@@ -89,25 +93,28 @@ private void handleSlimefunArmor(Player p, ItemStack[] armor, HashedArmorpiece[]
     }
 
     private void checkForSolarHelmet(Player p) {
-        // Temporary performance improvement
-        if (!SlimefunUtils.isItemSimilar(p.getInventory().getHelmet(), SlimefunItems.SOLAR_HELMET, true)) {
+        ItemStack helmet = p.getInventory().getHelmet();
+
+        if (SlimefunPlugin.getRegistry().isBackwardsCompatible() && !SlimefunUtils.isItemSimilar(helmet, SlimefunItems.SOLAR_HELMET, true, false)) {
+            // Performance saver for slow backwards-compatible versions of Slimefun
             return;
         }
 
-        SlimefunItem item = SlimefunItem.getByItem(p.getInventory().getHelmet());
+        SlimefunItem item = SlimefunItem.getByItem(helmet);
 
-        if (item instanceof SolarHelmet && Slimefun.hasUnlocked(p, item, true) && hasSunlight(p)) {
-            ItemEnergy.chargeInventory(p, (float) ((SolarHelmet) item).getChargeAmount());
+        if (item instanceof SolarHelmet && Slimefun.hasUnlocked(p, item, true)) {
+            ((SolarHelmet) item).rechargeItems(p);
         }
     }
 
     private boolean hasSunlight(Player p) {
-        return (p.getWorld().getTime() < 12300 || p.getWorld().getTime() > 23850) && p.getEyeLocation().getBlock().getLightFromSky() == 15;
+        World world = p.getWorld();
+        return (world.getTime() < 12300 || world.getTime() > 23850) && p.getEyeLocation().getBlock().getLightFromSky() == 15;
     }
 
     private void checkForRadiation(Player p) {
         // Check for a Hazmat Suit
-        if (!SlimefunUtils.isItemSimilar(SlimefunItems.SCUBA_HELMET, p.getInventory().getHelmet(), true) || !SlimefunUtils.isItemSimilar(SlimefunItems.HAZMAT_CHESTPLATE, p.getInventory().getChestplate(), true) || !SlimefunUtils.isItemSimilar(SlimefunItems.HAZMAT_LEGGINGS, p.getInventory().getLeggings(), true) || !SlimefunUtils.isItemSimilar(SlimefunItems.RUBBER_BOOTS, p.getInventory().getBoots(), true)) {
+        if (!SlimefunUtils.isItemSimilar(p.getInventory().getHelmet(), SlimefunItems.SCUBA_HELMET, true) || !SlimefunUtils.isItemSimilar(p.getInventory().getChestplate(), SlimefunItems.HAZMAT_CHESTPLATE, true) || !SlimefunUtils.isItemSimilar(p.getInventory().getLeggings(), SlimefunItems.HAZMAT_LEGGINGS, true) || !SlimefunUtils.isItemSimilar(p.getInventory().getBoots(), SlimefunItems.RUBBER_BOOTS, true)) {
```

Comment:
:LULW:

Reply:
??

Reply:
Wrong order. It was funny.

Reply:
Yeah, god knows for how long this must have been there...